futures-util/src/future/future/shared.rs

```diff
@@ -107,6 +116,15 @@ where
         }
         None
     }
+
+    /// Returns [`Some`] containing a [`WeakShared`] that may later be upgraded back to this [`Shared`]
+    /// or [`None`] if it has already been polled to completion.
```

Comment:
```suggestion
    /// Creates a new [`WeakShared`] to this [`Shared`].
    ///
    /// Returns [`None`] if it has already been polled to completion.
```

Reply:
I ultimately went with "for this [`Shared`]" here.